src/Container/ApplicationFactory.php

```diff
@@ -119,16 +122,16 @@ class ApplicationFactory
      */
     public function __invoke(ContainerInterface $services)
     {
-        $router = $services->has('Zend\Expressive\Router\RouterInterface')
-            ? $services->get('Zend\Expressive\Router\RouterInterface')
+        $router = $services->has(RouterInterface::class)
+            ? $services->get(RouterInterface::class)
             : new AuraRouter();
 
-        $finalHandler = $services->has('Zend\Expressive\FinalHandler')
-            ? $services->get('Zend\Expressive\FinalHandler')
+        $finalHandler = $services->has(FinalHandler::class)
```

Comment:
In this case, no; the final handler does not need to be an instance of `Zend\Stratigility\FinalHandler`; it can be a simple `callable`. This is the reason why the tests fail currently.

The reason I used `Zend\Expressive` for the namespace is because we have some plans to offer an alternate final handler solution that will provide templated error pages. However, again, we wouldn't want to typehint on that class, as any callable is allowed in the signature that consumes the service.

In other words, it's a service name that doesn't map 1:1 with a class.